### PR TITLE
Python notebook - displaying graphs

### LINKED ISSUE #72: Enable displaying graph images without saving to file in python notebooks

### PR DESCRIPTION
**Pull request by Kaasiak**
Makes filename optional and enables displaying the graphs in Jupyter notebooks without saving to a file.

Fixes a related unit test.

Resolves #72

### CODE REVIEW

**Comment by Kaasiak** [comment]
Before merging the same should be done for the ceg class. But waiting for the review first.

**Comment by g-walley** [comment]
@Kaasiak Cool, I've added a comment too

**Review by g-walley — COMMENTED** [review]
Sorry, I had started the review but left it pending! Just one comment really!

**Review by g-walley — COMMENTED** [review]
Looks great, thanks a lot!

**Comment by g-walley** [comment]
> Before merging the same should be done for the ceg class. But waiting for the review first.

Yes please, do this and we can merge

File: src/cegpy/trees/event.py [hunk 1]
```diff
@@ -357,22 +357,28 @@ def _generate_dot_graph(self, fill_colour=None):
...
+    
+        if get_ipython is not None:
             logger.info("--- Exporting graph to notebook ---")
-            return Image(graph.create_png())
+            graph_image = Image(graph.create_png())
+
+        return graph_image
+
+    def create_figure(self, filename=None):
+        return self._create_figure(self.dot_event_graph, filename)
```
[the remark this answers is not in the document]
Comment (Kaasiak) [reply src/cegpy/trees/event.py:381]: The EventTree class passes `self.dot_event_graph` to the `_create_figure()` method
Then, the StagedTree class can use the parent ` _create_figure()` but with `self.dot_staged_graph`

**Pull request merged**